secure_fw/CMakeLists.txt

```diff
@@ -331,8 +331,15 @@ function(set_up_secure_fw_build)
 		embedded_set_target_link_defines(TARGET ${EXE_NAME} DEFINES "TFM_MULTI_CORE_TEST")
 	endif()
 
+	#Collect executables to common location: <build_dir>/install/outputs/
+	if (DEFINED SECURE_AXF_DIR_PREFIX)
+		set(MY_BINARY_DIR ${SECURE_AXF_DIR_PREFIX})
+	else()
+		set(MY_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
+	endif()
+
 	#Generate binary file from executable
-	compiler_generate_binary_output(${EXE_NAME})
+	compiler_generate_binary_output(${EXE_NAME} ${MY_BINARY_DIR})
```

Comment:
If I understand the TF-M build system correctly, then it is wrong to directly target `${MY_BINARY_DIR}` for output.

Everything should be placed in the normal build path, and then be copied from their to the install folder, as part of the install target. That is what would happen on `make/ninja install`

Reply:
I'm pretty sure I'm following convention. AFAIU `MY_BINARY_DIR` and `SECURE_AXF_DIR_PREFIX` are used to divert the build output, not the install output. When I build it with ConfigRegression.cmake which sets `CORE_TEST` , the axf is not in its regular build location (secure_fw), but is instead in unit_test.

From secure_fw/CMakeLists.txt:
```
if(CORE_TEST)
	set(SECURE_AXF_DIR_PREFIX "${CMAKE_BINARY_DIR}/unit_test/")
	set_target_properties(${EXE_NAME} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${SECURE_AXF_DIR_PREFIX})
	embedded_set_target_link_defines(TARGET ${EXE_NAME} DEFINES "TFM_PARTITION_TEST_CORE")
endif()

...

#Collect executables to common location: <build_dir>/install/outputs/
if (DEFINED SECURE_AXF_DIR_PREFIX)
	set(MY_BINARY_DIR ${SECURE_AXF_DIR_PREFIX})
else()
	set(MY_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
endif()

...

install(FILES ${MY_BINARY_DIR}/${EXE_NAME}.axf
	      ${MY_BINARY_DIR}/${EXE_NAME}.bin
	      ${MY_BINARY_DIR}/${EXE_NAME}.hex # Added by us
	      ${MY_BINARY_DIR}/${EXE_NAME}.elf # Added by us
	      DESTINATION outputs/${TARGET_PLATFORM}/)
```

Reply:
ahh, right, my mistake.
The install destination is `output/${TARGET_PLATFORM}`, not `${MY_BINARY_DIR}`.